src/main/scala/com/fulcrumgenomics/bam/FindTechnicalReads.scala

```diff
@@ -77,33 +93,66 @@ class FindTechnicalReads
       val r1 = iter.next()
       val r2 = if (iter.hasNext && iter.head.getReadName == r1.getReadName) Some(iter.next()) else None
       val recs = Seq(Some(r1), r2).flatten
-      if (recs.exists(r => matcher.matches(r.getReadBases))) {
-        recs.foreach(out.addAlignment)
-        n += recs.size
+
+      val isTechnical = if (isUnmappedTemplate(r1)) {
+        val hit      = recs.toStream.flatMap(r => matcher.findMatch(r.getReadBases)).headOption
```

Comment:
Also, since you care only about the index of the hit, why not just have a method on matcher `findMatchIndex`?